gerrit.go

```diff
@@ -120,6 +120,9 @@ func NewClient(endpoint string, httpClient *http.Client) (*Client, error) {
 	if err != nil {
 		return nil, err
 	}
+	if !strings.HasSuffix(baseURL.Path, "/") {
```

Comment:
Would this change make the comment `// It must have a trailing slash.` above obsolete?

Reply:
Do you mean the one above the `baseURL` field of `Client` struct?

```Go
// baseURL is the base URL of the Gerrit instance for API requests.
// It must have a trailing slash.
baseURL *url.URL
```

If so, that comment still needs to be there, and it's very important. It documents the fact that the value of `baseURL` must always have a trailing slash.

This code here is a part of `NewClient` which takes a `gerritURL string` parameter, and it's documented as follows:

```Go
// ...
// If gerritURL does not have a trailing slash, one is added automatically.
// ...
func NewClient(gerritURL string, httpClient *http.Client) (*Client, error) {
```

So `gerritURL` may not have a trailing slash, and we want to add one here in `NewClient` before assigning it to `Client.baseURL` field, otherwise we'd be violating the requirement of that field.

Given that the `Client.baseURL` field is now guaranteed to have a trailing slash, it allowed me to remove the following code from the `buildURLForRequest` method, because it was no longer necessary:

```diff
 	u := c.baseURL.String()
-
-	// If there is no / at the end, add one
-	if strings.HasSuffix(u, "/") == false {
-		u += "/"
-	}
```

We're making use of `baseURL`'s `// It must have a trailing slash.` property there.

Does that make sense @andygrunwald?

Reply:
Make total sense. I forgot the fact that baseURL of the client struct is not exported and only accessible through the getter function.

Reply:
> I forgot the fact that baseURL of the client struct is not exported and only accessible through the getter function.

As of this PR. It wasn't accessible before, which is what I wanted to fix. :)